network_test.go

```diff
@@ -185,3 +185,50 @@ func TestUpdateRoutes(t *testing.T) {
 	assert.True(t, reflect.DeepEqual(results.Routes[0], testRoutes.Routes[1]),
 		"Interface created didn't match: got %+v, expecting %+v", results.Routes[0], testRoutes.Routes[1])
 }
+
+func TestListInterfaces(t *testing.T) {
+	skipUnlessRoot(t)
+
+	s := sandbox{}
+
+	ifc := pb.Interface{
+		Name:   "enoNumber",
+		Mtu:    1500,
+		HwAddr: "02:00:ca:fe:00:48",
+	}
+	ip := pb.IPAddress{
+		Family:  0,
+		Address: "192.168.0.101",
+		Mask:    "24",
+	}
+	ifc.IPAddresses = append(ifc.IPAddresses, &ip)
+
+	netHandle, _ := netlink.NewHandle()
+	defer netHandle.Delete()
+
+	// create a dummy link that we can test update on
+	macAddr := net.HardwareAddr{0x02, 0x00, 0xCA, 0xFE, 0x00, 0x48}
+	link := &netlink.Dummy{
+		LinkAttrs: netlink.LinkAttrs{
+			MTU:          1500,
+			TxQLen:       -1,
+			Name:         "ifc-name",
+			HardwareAddr: macAddr,
+		},
+	}
+	netHandle.LinkAdd(link)
+	netHandle.LinkSetUp(link)
+
+	resultingIfc, err := s.updateInterface(netHandle, &ifc)
+	assert.Nil(t, err, "Unexpected update interface failure: %v", err)
```

Comment:
Nit: I find it's much cleaner to avoid having to pass `t` to all the assert function using the following idiom:

```
assert := assert.New(t)
   :

assert.Nil(err, "Unexpected update interface failure: %v", err)
```

Reply:
Yes, definitely. Fixed